metron-stellar/stellar-common/src/test/java/org/apache/metron/stellar/dsl/functions/DateFunctionsTest.java

```diff
@@ -182,7 +182,8 @@ public void testDayOfMonthNull() {
   @Test
   public void testWeekOfYear() {
     Object result = run("WEEK_OF_YEAR(epoch)");
-    assertEquals(35, result);
+    calendar.setTimeInMillis(AUG2016);
```

Comment:
it it save to morph/mutate an object that is shared? What about a short lived object like that:
   @Test
   public void testWeekOfYear() {
     Object result = run("WEEK_OF_YEAR(epoch)");
-    assertEquals(35, result);
+    Calendar testCalendar = Calendar.getInstance();
+    testCalendar.setTimeInMillis(AUG2016);
+    assertEquals(testCalendar.get(Calendar.WEEK_OF_YEAR), result);
   }

Reply:
It is perfectly safe: The instance is create in the @Before annotated method, so in fact, I am already doing exactly what you suggest.  See: https://github.com/simonellistonball/metron/blob/bafd3827d273c5c33b4293fd338595b8de3a75b2/metron-stellar/stellar-common/src/test/java/org/apache/metron/stellar/dsl/functions/DateFunctionsTest.java#L61 

So there is no sharing of this object between tests, since a new instance is acquired before each test method is run. 

The second issue you raise should be handled by a separate unit of change and a separate ticket if it is really necessary (not sure why it would be, but perhaps you could raise a JIRA and explain).